feature/1 5solving fixed mistakes

Val28He opened the pull request:
Исправлены ошибки

Val28He committed:
Update tr.md

Сами изменения были сделаны 26.09

Val28He committed:
Done

Исправлены ошибки

Val28He committed:
Update tr.md

Заполнены оставшиеся пункты технического решения

SkyWaet commented on docs/tr.md:
```diff
@@ -1,44 +1,163 @@
 ## Технические сценарии
 
----
+* I. Обновление информации об изображении, привязанному к определённому серверу.
```
Не уверен, что такой сценарий реально реализовать без протокола WebSocket. REST API тут точно не поможет - он рассчитан на взаимодействие в формате запрос-ответ, и потому сервер по своей инициативе клиенту отправить ничего не может. Отразите это на диаграмме компонентов, пожалуйста

SkyWaet commented on docs/tr.md:
```diff
@@ -1,44 +1,163 @@
+  + 2. Клиентское приложение отправляет информацию об изменении API Gateway.
+  + 3. API Gateway отправляет эту информацию Change Information Handler.
+  + 4. Change Information Handler отправляет эту информацию на нужный Processing Server.
+  + 5. Processing Server обрабатывает изменения и применяет их к изображению.
```
Поскольку у вас одновременно в рамках одной сессии могут работать несколько пользователей, они вполне могут случайно внести в изображение конфликтующие правки - что-то нарисовать в одном и том же месте. Стоит подумать, как именно вы такие ситуации будете разрешать и отразить это в сценарии

SkyWaet commented on docs/tr.md:
```diff
@@ -1,44 +1,163 @@
 
----
+* API Gateway - входная точка в систему, отвечает за маршрутиризацию криентских запросов.
+* Server Manager - отвечает за наблюдение за состоянием серверов и распределение пользователей по ним. 
+* Processing Server - Отвечает за обработку изменений в изображениях.
```
Правильно ли я понимаю, что Server Manager и Processing Server - stateful компоненты, которые все данные о сессиях / изображениях хранят у себя в памяти?

SkyWaet reviewed (changes requested):
Не хватает диаграмм: диаграммы компонентов + sequence диаграмм для каждого из технических сценариев. 

В последнем разделе не хватает DoD

Val28He committed:
Update tr.md

Исправлен DoD , разделён план разработки тестирования. Также добавлена диаграмма компонентов

Val28He committed:
Update tr.md

Добавлен DoD для расширенной части

Val28He committed:
Update tr.md

Добавлены диаграммы для сценариев

SkyWaet reviewed (changes requested):
Drawio диаграмму еще логично вставлять ссылкой, потому что в формате png ее комментировать невозможно, а в xml никто не поймет, что нарисовано. mermaid же ценен ровно тем, что его можно вставить в markdown как исходный код, и github самостоятельно ее отобразит. И не нужно будет по ссылкам на сторонние сайты переходить. Пожалуйста, посмотрите, как диаграммы подключены у меня в примере, и сделайте так же: https://github.com/spbu-ds-practicum-2025/example-project/blob/main/docs/tr.md?plain=1#L124

Val28He committed:
Update tr.md

Изменены диаграммы, теперь они отображаются корректно

Val28He committed:
Update tr.md

Теперь есть коды в диаграммах

Val28He committed:
Update tr.md

Исправлены некоторые диаграммы
